File: CRM.JFLEAD.Domain/Entities/Lead.cs
using CRM.SharedKernel.Domain;

namespace CRM.JFLEAD.Domain
{

    public class Lead : BaseEntity
    {
        public LeadType Type { get; set; }
        public string? Denomination { get; set; } // For Societe
        public string? Civilite { get; set; } // For Individu
        public string? Nom { get; set; }
        public string? Prenom { get; set; }
        public string? Adresse { get; set; }
        public string? CodePostale { get; set; }
        public string? Ville { get; set; }
        public string? Region { get; set; }

        public string? Pays { get; set; }
        public string? Telephone { get; set; }

        public string? Fax { get; set; }
        public string? Email { get; set; }
        public string? SiteWeb { get; set; }
        public string? Categorie { get; set; }
        public string? SecteurActivite { get; set; }
        public string? Provenance { get; set; }
        public LeadStatus Status { get; set; }
        public int AssignedTo { get; set; }
    }
}

File: Front/src/app/core/Models/Lead.ts
export enum LeadType {
  Societe = 0,
  Individu = 1
}

export enum LeadStatus {
  Nouveau = 0,
  Assigne = 1,
  EnCours = 2,
  ConvertiGagne = 3,
  ConvertiPerdu = 4
}

export interface Lead {
  id: string;                // Corresponding to BaseEntity's Id property
  type: LeadType;            // Enum or string, depending on how you define LeadType in your Angular app
  denomination?: string;     // For Societe
  civilite?: string;         // For Individu
  nom?: string;              // Last name
  prenom?: string;           // First name
  adresse?: string;          // Address
  codePostale?: string;      // Postal code
  ville?: string;            // City
  region?: string;           // Region
  pays?: string;             // Country
  telephone?: string;        // Phone number
  fax?: string;              // Fax number
  email?: string;            // Email address
  siteWeb?: string;          // Website
  categorie?: string;        // Category
  secteurActivite?: string;  // Industry sector
  provenance?: string;       // Origin
  status: LeadStatus;        // Enum or string, depending on how you define LeadStatus in your Angular app
  assignedTo: number;        // Assigned collaborator's ID
}
